src/test/java/space/obminyashka/items_exchange/end2end/AdvertisementFlowTest.java

```diff
@@ -226,6 +227,23 @@ void createAdvertisement_shouldCreateValidAdvertisement() throws Exception {
         assertEquals(6, advertisementRepository.count());
     }
 
+    @Test
+    @WithMockUser(username = "admin")
+    @DataSet("database_init.yml")
```

Comment:
You can also validate the DB content verifying that any new advertisement wasn't created by using **ExpectedDataSet("database_init.yml")**